mlx-macros/tests/test_generate_builder.rs

```diff
@@ -0,0 +1,29 @@
+use mlx_macros::GenerateBuilder;
+
+#[derive(Debug, GenerateBuilder)]
+struct TestStruct {
+    #[optional(default_value = TestStruct::DEFAULT_OPT_FIELD_1)]
+    opt_field_1: i32,
+    #[optional(default_value = TestStruct::DEFAULT_OPT_FIELD_2)]
```

Comment:
Do you think it's still useful to have consts for defaults here? The fact that we can annotate means we can quickly see what those defaults are going to be without the indirection. Or are you worried about perf?

Reply:
Also because this is a macro, IDEs won't always be able to CMD+click into the definition.